feat: add support for preferences in ReactiveBase (#323)
appbaseio/Docs: content/docs/reactivesearch/v3/overview/reactivebase.md

    > Mandatorily pass ***`mapLibraries={['places']}`*** when using either or both of GeoDistanceDropdown/ GeoDistanceSlider components from [ReactiveMaps 🗺️ ](https://docs.appbase.io/docs/reactivesearch/v3/overview/reactivemaps/).

-   **preferences** `Object` [optional]

>     accepts an object to configure the search settings for components. The `preferences` property enables you to configure the search settings for your components at one place. The preferences object must follow the following structure:

Reviewer: ```suggestion
    accepts an object to configure the search settings for components. The `preferences` property allows configuring the search settings for your components in one place. The preferences object must follow the following structure:

```

```ts
{
    componentSettings: {
        [componentId]: {
            // If disabled, the component would not get rendered and would not constitute the search query
            enabled: true,
            rsConfig: {
                // props supported by reactivesearch components
            }
            // can have additional keys to store meta data for components
            // the preference object for a component can be accessed using the 
            // `SearchPreferencesContext` context.
            // For e.g custom property to control the collapsible property for a facet
            isCollapsible: true,
        }
    }
}
```

> To connect a reactivesearch component to a preference use the similar component Id prop. The following example has defined the `preferences` object for `bookSearch` component, the `DataSearch` component is using the same component Id (`bookSearch`).

Reviewer: ```suggestion
To connect a ReactiveSearch component to a preference, use the `componentId` prop. The following example has defined the `preferences` object for `bookSearch` component, the `DataSearch` component is using the same componentId (`bookSearch`).
```

```jsx
    <ReactiveBase preferences={{
        componentSettings: {
            bookSearch: {
                rsConfig: {
                    dataField: 'original_title',
                    title: 'Search for Books',
                    size: 5,
                }
            }
        }
    }}>
        <DataSearch componentId="bookSearch" />
    </ReactiveBase>
```

> Additionally, the reactivesearch components support `preferencesPath` prop which is helpful to define the path of preference object for a component. It is helpful when you have to use conflicting component Ids. The following example defines the preferences for `home` and `search` pages, components have defined the `preferencesPath` prop to connect to preferences.

Reviewer: ```suggestion
Additionally, the ReactiveSearch components support `preferencesPath` prop which is helpful to define the path of preference object for a component. It is helpful when you have to use conflicting component Ids. The following example defines the preferences for `home` and `search` pages, components have defined the `preferencesPath` prop to connect to preferences.
```

```jsx
    <ReactiveBase preferences={{
        pages: {
            home: {
                bookSearch: {
                    rsConfig: {
                        dataField: 'original_title',
                        title: 'Search for Books',
                        size: 5,
                    }
                }
            },
            search: {
                bookSearch: {
                    rsConfig: {
                        dataField: ['original_title', 'authors', 'publishers'],
                        title: 'Search for Books, Authors, Publishers',
                        size: 10,
                    }
                }
            }
        }
    }}>
        {/** home page */}
        <DataSearch 
            preferencesPath="pages.home.bookSearch" 
            componentId="bookSearch" 
        />
         {/** search page */}
        <DataSearch 
            preferencesPath="pages.search.bookSearch" 
            componentId="bookSearch" 
        />
    </ReactiveBase>
```
> Note:
>

> > Preferences is meant to be one time configuration for components. We don't recommend to mutate it because it can cause performance issues.

Reviewer: ```suggestion
> Preferences is meant to be a one time configuration for components. We don't recommend to mutate it as it can cause performance issues.
```
